new addon - cookienotice - configure, show and handle a simple cookie…

Ixiter opened the pull request:
similar to the pagheheader addon, an admin user can configure a cookie usage notice which will be shown until the user confirms by clicking an OK Button.
Cookie usage notice and OK button text are configurable.
Handling of the cookie accept cookie is done by the addon.

You can see the addon in action at my https://socivitas.com (german)

tobiasd commented:
I have another question (though the base messages.po generation would be nice)--what license do you have in mind for your addon?

tobiasd commented on cookienotice/README:
```diff
@@ -0,0 +1,7 @@
+
+Configure, show and handle a simple cookie usage notice.
+
+Author: Peter liebetrau <https://socivitas.com/profile/peerteer>
```
If you like, you can also make a Markdown formatted README.md file.

And TBH I would not really know what you mean with cookie usage notice, if I had not checked your profile.

Ixiter replied:
Ah great. Saw README.md with themes already. 
hmm .. the cookie usage notice  ... how else to call it ? ..

tobiasd replied:
I would describe it as absolute annoying but eventually necessary notification about the usage of cookies. This kind of things you klick ok on but don't read.

tobiasd replied:
Or, you could add a screen shot of such a message to the README.md file. Maybe this would spare a 1000 words description of the thing.

Ixiter replied:
done - used your perfect description

Ixiter replied:
didnt change to .md yet. Maybe with the next version

tobiasd commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="/addon/cookienotice/cookienotice.css" media="all" />' . "\r\n";
+
+
+	$text = Config::get('cookienotice', 'text');
```
You can pass a default value as 3rd parameter to the `Config::get` call.

Ixiter replied:
Ah, thats cool. Thank you.

Ixiter replied:
done

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+function cookienotice_install()
+{
+	$file = 'addon/cookienotice/cookienotice.php';
+	Addon::registerHook('page_content_top', $file, 'cookienotice_page_content_top');
```
In the upcoming release this method has been deprecated in favor of `Hook::register`. Since we are so close from the release date I will ask you to update all calls from `Addon::registerHook` to `Hook::register` and `Addon::unregisterHook` to `Hook::unregister`.

Ixiter replied:
Thank you. Saw that already in the develop branch in other addons. I will change it.

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+
+function cookienotice_addon_settings(&$a, &$s)
+{
+	if (!is_site_admin())
```
Code standards: Please add braces around all conditional structures.

Ixiter replied:
I' with you and I also dont like "missing braces". Since I "stole" this code from the pageheader plugin, I am not that guilty. :D
Will fix these things.

Ixiter replied:
done

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+
+function cookienotice_addon_settings_post(&$a, &$b)
+{
+	if (!is_site_admin())
```
Code standards: Please add braces around all conditional structures.

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+	Addon::unregisterHook('addon_settings_post', $file, 'cookienotice_addon_settings_post');
+}
+
+function cookienotice_addon_settings(&$a, &$s)
```
Please update your code to
```suggestion
function cookienotice_addon_settings(App $a, &$s)
```

Ixiter replied:
done - had to add the complete namespace for some reason.

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+	return;
+}
+
+function cookienotice_addon_settings_post(&$a, &$b)
```
Please update your code to
```suggestion
function cookienotice_addon_settings_post(App $a, &$b)
```

Ixiter replied:
done

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+ * @param App $a
+ * @param string $b - The page html before page_content_top
+ */
+function cookienotice_page_content_top($a, &$b)
```
Please update your code to
```suggestion
function cookienotice_page_content_top(App $a, &$b)
```

Ixiter replied:
done

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,107 @@
+ * @param App $a
+ * @param string $b - The page html
+ */
+function cookienotice_page_end($a, &$b)
```
Please update your code to
```suggestion
function cookienotice_page_end(App $a, &$b)
```

Ixiter replied:
done

MrPetovan commented on cookienotice/templates/cookienotice.tpl:
```diff
@@ -0,0 +1,23 @@
+<style type="text/css">
+	#cookienotice-box {
```
Please move the CSS in the `cookienotice.css` file.

Ixiter replied:
done

MrPetovan commented on cookienotice/templates/head.tpl:
```diff
@@ -0,0 +1,40 @@
+<!-- <link rel="stylesheet" type="text/css" href="/addon/cookienotice/css/cookienotice.css" /> -->
```
Please remove commented out code.

Ixiter replied:
done

MrPetovan commented on cookienotice/templates/head.tpl:
```diff
@@ -0,0 +1,40 @@
+<!-- <link rel="stylesheet" type="text/css" href="/addon/cookienotice/css/cookienotice.css" /> -->
+<script>
+	window.addEventListener("load", function () {
```
Please move the Javascript code in a new `cookienotice.js` file.

Ixiter replied:
done

Ixiter commented:
i guess I fixed all from the reviews.
Also added docblocks to all functions.
I learned a lot about the codes. It was fun.

Any other hints|complaints|tipps|tricks ?

MrPetovan commented on cookienotice/cookienotice.js:
```diff
@@ -0,0 +1,37 @@
+	if (cookie == "") {
+		document.getElementById('cookienotice-box').style.display = 'block';
+		document.getElementById('cookienotice-ok-button').onclick = function () {
+			console.log('clicked');
```
Please remove debug code from committed code.

Ixiter replied:
no problem with second pass. I am still learning.

done

MrPetovan replied:
The `console.log` call is still present in the last version of this PR.

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,151 @@
+ * Author: Peter Liebetrau <https://socivitas/profile/peerteer>
+ * 
+ */
+use Friendica\Core\Addon;
```
This `use` statement doesn't seem to be used anymore.

Ixiter replied:
done

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,151 @@
+
+	/* Add our stylesheet to the page so we can make our settings look nice */
+
+	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="/addon/cookienotice/cookienotice.css" media="all" />' . "\r\n";
```
Please use `App::registerStylesheet` instead.

Ixiter replied:
done

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,151 @@
+ */
+function cookienotice_page_content_top(\Friendica\App $a, &$b)
+{
+	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="/addon/cookienotice/cookienotice.css" media="all" />' . "\r\n";
```
Please use `App::registerStylesheet` instead.

Ixiter replied:
done

MrPetovan commented on cookienotice/templates/head.tpl:
```diff
@@ -0,0 +1 @@
+<script type="text/javascript" src="/addon/cookienotice/cookienotice.js"></script>
```
Please use `App::registerFooterScript` instead.

Ixiter replied:
done

tobiasd commented on cookienotice/cookienotice.php:
```diff
@@ -60,9 +60,8 @@ function cookienotice_addon_settings(\Friendica\App $a, &$s)
-
-	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="/addon/cookienotice/cookienotice.css" media="all" />' . "\r\n";
-
+    $stylesheetPath = 'addon/cookienotice/cookienotice.css';
```
Could you please convert the spaces to TABS

Ixiter replied:
arrgh .. I dont know why, but my netbeans does it kinda randomly.

fixed

tobiasd commented on cookienotice/cookienotice.php:
```diff
@@ -120,9 +119,11 @@ function cookienotice_addon_settings_post(\Friendica\App $a, &$b)
-	$a->page['htmlhead'] .= '<link rel="stylesheet"  type="text/css" href="/addon/cookienotice/cookienotice.css" media="all" />' . "\r\n";
-	$head = file_get_contents(__DIR__ . '/templates/head.tpl');
-	$a->page['htmlhead'] .= $head;
+    $stylesheetPath = 'addon/cookienotice/cookienotice.css';
```
Here as well, please use TABS instead of spaces.

MrPetovan commented on cookienotice/cookienotice.php:
```diff
@@ -0,0 +1,145 @@
+ */
+function cookienotice_install()
+{
+	$file = 'addon/cookienotice/cookienotice.php';
```
Tip: This string is actually contained in the magic constant `__FILE__`. If you ever rename your file for whatever reason, this constant will update accordingly.

Ixiter commented:
changes done

Ixiter commented:
Yeah, great. Thank you for you reviews. I learned a lot.
